components/Search/SearchBox.tsx

```diff
@@ -32,7 +31,9 @@ export const SearchBox = ({
     <>
       <div className="rs-mb-4">
         <SearchForm
-          defaultValue={searchTerm || undefined}
+          // Ensure the form re-renders when query changes thus updating the input value accordingly
+          key={query ?? ''}
```

Comment:
Previously, if the search page input bar already has a query term in it, doing a search in the modal doesn't update the page's search bar. Adding this force the page input bar to update to the active query that was entered in the modal.